### PR TITLE
PP-591: pbs_mom not coming up after switching on compute node on cpus…

### PR DESCRIPTION
**Pull request by nithinj**
…et machines

#### Issue-ID

* **[PP-591](https://pbspro.atlassian.net/browse/PP-591)**

#### Problem description
* pbs_mom not coming up after switching on compute node on cpuset machines

#### Cause / Analysis
* systemctl preset will disable services which does not have a preset file(In fedora)

#### Solution description
* Introduced a preset file.
* Updated condition for checking the presence of systemd
* Updated sed which modifies unit file
* Excluded unit file in pbspro.spec in systems which will not support systemd in order to avoid packaging error. 

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/cmds/scripts/pbs_postinstall.in [hunk 1]
```diff
@@ -384,18 +384,21 @@ install_pbsinitd() {
...
-				eval "sed -i 's;\(^[[:space:]]*ExecStop=\)[^[:space:]]*;\1${PBS_EXEC}/libexec/pbs_init.d stop;' \"$unitfilepath/pbs.service\""
-				if [ -x /bin/systemctl ] ; then
-					/bin/systemctl daemon-reload && /bin/systemctl enable pbs
-					[ $? != 0 ] && echo "Failed to register PBS Pro as a service in systemd"
+
+			pbs_unitfile="/usr/lib/systemd/system/pbs.service"
+			if [ -f "${pbs_unitfile}" ]; then
+				presetdir="/usr/lib/systemd/system-preset"
+				eval "sed -i 's;\(^[[:space:]]*SourcePath=\)[^[:space:]]*;\1${PBS_EXEC}/libexec/pbs_init.d;' \"$pbs_unitfile\""
+				eval "sed -i 's;\(^[[:space:]]*ExecStart=\)[^[:space:]]*;\1${PBS_EXEC}/libexec/pbs_init.d;' \"$pbs_unitfile\""
```
Comment (mike0042) [inline src/cmds/scripts/pbs_postinstall.in]: Why was "start" removed on this line?

File: src/cmds/scripts/pbs_postinstall.in [hunk 2]
```diff
@@ -384,18 +384,21 @@ install_pbsinitd() {
...
+				eval "sed -i 's;\(^[[:space:]]*ExecStop=\)[^[:space:]]*;\1${PBS_EXEC}/libexec/pbs_init.d;' \"$pbs_unitfile\""
```
Comment (mike0042) [inline src/cmds/scripts/pbs_postinstall.in]: Why was "stop" removed on this line?

Comment (nithinj) [inline src/cmds/scripts/pbs_postinstall.in]: The unit file in src/scripts/pbs.service contains the start/stop commands. The script used to append one additional start/stop.

Comment (mike0042) [inline src/cmds/scripts/pbs_postinstall.in]: I see that now, thanks. So autoconf is going to perform the substitution of @prefix@ in pbs.service.in. Why does pbs_postinstall modify this file again at install time?

Comment (nithinj) [inline src/cmds/scripts/pbs_postinstall.in]: The prefix substituted by autoconf can get modified during installation using either `--prefix or --rellocate` commands. This value will be used as the new PBS_EXEC in pbs.service file.

Comment (mike0042) [inline src/cmds/scripts/pbs_postinstall.in]: Thank you for the explanation. Changes approved.

**Comment by mike0042** [comment]
Please update your branch in preparation for merge.

**Comment by nithinj** [comment]
I've updated the branch. Please check.

**Comment by mike0042** [comment]
I merged PP-617 a few minutes ago (before I saw your message), so I'll have to ask that you update your branch again.

**Comment by nithinj** [comment]
I've updated the branch. @PBSPro/pbspro-maintainers please take it forward.

**Pull request merged**